src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java

```diff
@@ -61,28 +63,44 @@
 	 *  display width will be based on 3 digits.
 	 *
 	 *  @param component  the related text component
+	 *  @param left_to_right  whether the text is left to right, this
+	 *                        affects alignment of line numbers and the
+	 *                        border of the panel
 	 */
-	public TextLineNumber(JTextComponent component)
+	public TextLineNumber(JTextComponent component, boolean left_to_right)
 	{
-		this(component, 3);
+		this(component, left_to_right, 3);
 	}
 
 	/**
 	 *	Create a line number component for a text component.
 	 *
 	 *  @param component  the related text component
+	 *  @param left_to_right  whether the text is left to right, this
+	 *                        affects alignment of line numbers and the
+	 *                        border of the panel
 	 *  @param minimumDisplayDigits  the number of digits used to calculate
 	 *                               the minimum width of the component
 	 */
-	public TextLineNumber(JTextComponent component, int minimumDisplayDigits)
+	public TextLineNumber(JTextComponent component, boolean left_to_right,
+			      int minimumDisplayDigits)
 	{
 		this.component = component;
+		if (left_to_right)
+		{
+			setDigitAlignment( RIGHT );
+			this.OUTER = new MatteBorder(0, 0, 0, 2, Color.GRAY);
+		}
+		else
+		{
+			setDigitAlignment( LEFT );
+			this.OUTER = new MatteBorder(0, 2, 0, 0, Color.GRAY);
```

Comment:
Just to understand - the "directions" (left, right, etc) in the MatteBorder constructor are absolute, right? They don't depend on what the orientation of the containing component is?

Reply:
I think so, the border looks right with this change, without it the border of the Arabic pane would be on the right.